tests/0122-ssl_cb.c

```
@@ -0,0 +1,308 @@
+        TEST_SKIP("Test requires SSL support\n");
+        goto cleanup;
+    }
+    if (!test_check_builtin("ssl_client_cert_callback")) {
+        TEST_SKIP("Test requires ssl_client_cert_callback support (i.e. built with openssl >= 1.0.2\n");
+        goto cleanup;
+    }
+
+
+    read_ssl_file_from_env(
+            "RDK_SSL_ca_pem", &ctx.ca_cert_pem, &ctx.ca_cert_pem_length
+    );
+    read_ssl_file_from_env(
+            "RDK_SSL_pub_pem", &ctx.direct_signed_client_cert_pem, &ctx.direct_signed_client_cert_pem_length
+    );
+    read_ssl_file_from_env(
+            "RDK_SSL_priv_pem", &ctx.direct_signed_client_key_pem, &ctx.direct_signed_client_key_pem_length
+    );
+    read_ssl_file_from_env(
+            "RDK_UNTRUSTEDSSL_pub_pem", &ctx.self_signed_client_cert_pem, &ctx.self_signed_client_cert_pem_length
+    );
+    read_ssl_file_from_env(
+            "RDK_UNTRUSTEDSSL_priv_pem", &ctx.self_signed_client_key_pem, &ctx.self_signed_client_key_pem_length
+    );
+    read_ssl_file_from_env(
+            "RDK_INTERMEDIATESSL_pub_pem", &ctx.intermediate_signed_client_cert_pem, &ctx.intermediate_signed_client_cert_pem_length
+    );
+    read_ssl_file_from_env(
+            "RDK_INTERMEDIATESSL_priv_pem", &ctx.intermediate_signed_client_key_pem, &ctx.intermediate_signed_client_key_pem_length
+    );
+    read_ssl_file_from_env(
+            "RDK_INTERMEDIATESSL_intermediate_pub_pem", &ctx.intermediate_signed_intermediate_cert_pem, &ctx.intermediate_signed_intermediate_cert_pem_length
+    );
+    read_ssl_file_from_env(
+            "RDK_INTERMEDIATESSL_intermediate_priv_pem", &ctx.intermediate_signed_intermediate_key_pem, &ctx.intermediate_signed_intermediate_key_pem_length
+    );
+    if (test_curr->state == TEST_SKIPPED || test_curr->state == TEST_FAILED) {
+        goto cleanup;
+    }
+
+    do_test_ssl_cb(&ctx, 1, ssl_client_cert_cb_good_direct);
+    if (test_curr->state == TEST_SKIPPED || test_curr->state == TEST_FAILED) {
```

Remove these, we want to fail fast, so use TEST_FAIL instead of FAIL_LATER.